ag_data/models.py

```diff
@@ -72,3 +72,39 @@ class AGMeasurement(models.Model):
 
 class AGActiveEvent(models.Model):
     agevent = models.ForeignKey(AGEvent, null=True, on_delete=models.SET_NULL)
+
+
+class ErrorLog(models.Model):
+    # declearation error
+    UNKNOWN_FMT = "UNKNOWN_FORMAT"
+    MISSING_COL = "MISSING_COLUMN"
+    MISSING_FIELD_IN_RAW = "MISSING_FIELD_IN_RAW_READING"
+    INVALID_COL_NM = "INVALID_COLUMN_NAME"
+    INVALID_COL_VL = "INVALID_COLUMN_VALUE"
+    INVALID_FIELD_IN_RAW = "INVALID_FIELD_IN_RAW_READING"
+    ERROR_F_PROC_MMT = "FORMULA_PROCESS_MEASUREMENT_ERROR"
+    EXTRA_KEYVAL_IN_MMT = "EXTRANEOUS_KEY_VALUE_PAIR_IN_MEASUREMENT"
+    OTHER = "OTHER_ERROR"
+
+    ERROR_CODE_CHOICES = [
+        (UNKNOWN_FMT, "Unknown Format"),
+        (MISSING_COL, "Missing Column"),
+        (MISSING_FIELD_IN_RAW, "Missing Field In Raw Reading"),
+        (INVALID_COL_NM, "Invalid Column Name"),
+        (INVALID_COL_VL, "Invalid Column Value"),
+        (INVALID_FIELD_IN_RAW, "Invalid Field In Raw Reading"),
+        (ERROR_F_PROC_MMT, "Error When Formula Processing Measurement"),
+        (EXTRA_KEYVAL_IN_MMT, "Extraneous Key-Value Pair In Measurement"),
+        (OTHER, "Other Error"),
+    ]
+
+    error_id = models.AutoField(primary_key=True)
+    error_timestamp = models.DateTimeField(default=timezone.now)
+    error_code = models.CharField(
+        max_length=30, choices=ERROR_CODE_CHOICES, default=OTHER
+    )
+    error_description = models.CharField(max_length=100, null=False, blank=False)
+    error_raw_data = models.CharField(max_length=500, null=False)
```

Comment:
I love that this model is exactly how we expected it to be. I also see there is prefix (`error_`) for each column in `ErrorLog`. I personally do not have any preference to this. If we look at other models in this file, we are not using prefixes any more. And I feel it would be more seamless if we dispense with the prefixes.

Reply:
Yes, we can make a consensus with other model names.

Reply:
ok, great!

Reply:
I will mark conversations as resolved when I review the updated pr.